core/controllers/topic_editor.py

```diff
@@ -83,6 +85,47 @@ def get(self, topic_id):
         self.render_template('pages/topic_editor/topic_editor.html')
 
 
+class EditableSubtopicPageDataHandler(base.BaseHandler):
+    """The data handler for subtopic pages."""
+
+    def _require_valid_version(
+            self, version_from_payload, subtopic_page_version):
+        """Check that the payload version matches the given subtopic page
+        version.
+        """
+        if version_from_payload is None:
+            raise base.BaseHandler.InvalidInputException(
+                'Invalid POST request: a version must be specified.')
+
+        if version_from_payload != subtopic_page_version:
+            raise base.BaseHandler.InvalidInputException(
+                'Trying to update version %s of subtopic page from version %s, '
+                'which is too old. Please reload the page and try again.'
+                % (subtopic_page_version, version_from_payload))
+
+    @acl_decorators.can_edit_topic
```

Comment:
Have this be more narrow -- just can_edit_subtopic_page. Reason is that this will likely be opened more widely in the future (like Wikipedia pages).

Reply:
Done.